### DISCUSSION TITLE
How to check if running in Teams

### DISCUSSION THREAD

**Discussion by svrooij** [microsoft/hack-together-teams#74] [Q&A] [answered]
I would assume the TeamsJS SDK would have some sort of **Are we running inside Teams** function, but the Blazor sample just uses the following javascript code:

```js
// Come from here: https://github.com/wictorwilen/msteams-react-base-component/blob/master/src/useTeams.ts
export function inTeams() {
    if (
        (window.parent === window.self && window.nativeInterface) ||
        window.navigator.userAgent.includes("Teams/") ||
        window.name === "embedded-page-container" ||
        window.name === "extension-tab-frame"
    ) {
        return true;
    }
    return false;
}
```

Is this the recommended way to check if the app is running in Teams or not?

Reason I'm asking, we see a lot more use cases where Team apps are loaded in SharePoint or at Office.com in the sidebar. Will all those other app loading options also result in `true` when this function is updated? Or maybe should there be something similar in the Teams SDK so everybody can call the same function and Microsoft has to be the one to keep it updated.

**Comment by waldekmastykarz** [comment]
It does indeed. You can use the `getContext` function to detect the host and then the `isSupported` function to detect specific capabilities that stem from the host. For more information see: https://learn.microsoft.com/en-us/microsoftteams/platform/tabs/how-to/using-teams-client-library?tabs=javascript%2Cmanifest-teams-toolkit

**Reply by svrooij** [reply]
So how do I test if this is supported `authentication.getAuthToken({...})`?
The `function isSupported()` only exists 41 times and it's not on all namespaces.

The question actually is: How do I test if the app is even run from any of the available hosts or if it's just opened as a regular webpage?

**Reply by waldekmastykarz** [reply, marked as answer]
If I'm not mistaken, when you try to get Teams context outside of Teams, it will give you null which indicates that you're not running inside of Teams. `isSupported` is meant for checking capabilities, like sending emails, chat messages etc, which is the recommended  way to determine the behavior inside your app because the functionality of the host could change in the future.
